### ISSUE TITLE
How to decode generic without "sniffing" the body?

### ISSUE THREAD

**Issue by crimsonvspurple** [google/json_serializable.dart#879]
https://github.com/google/json_serializable.dart/blob/824950daf076600a7666bc5d451e9997dfd94462/example/lib/generic_response_class_example.dart#L30

I started Dart only 3 days ago so I am still having difficulties wrapping my head around. But I am not seeing a clean way to do this with only generics?

I can pass a `T Function(Object body) bodyToJson` when dealing with the API call but that function can't be set in `@JsonKey(fromJson: bodyToJson)`.

In the example, a `BaseResponse<SomeEntity>()` should be able to return a `BaseResponse` which has `data` as `SomeEntity` without the need for content "inspecting" which does not work in practice.

**Comment by kevmoo** [comment]
Are you still having this issue? Can you provide some more context?

**Comment by crimsonvspurple** [comment]
> Are you still having this issue? Can you provide some more context?

Yes, I worked around the issue somehow but I believe I couldn't really achieve the niceties I'm used to in Java/Jackson.

My apologies. I forgot about this issue. I will go thru the code base and try to come up with more context/sample for you in a few days.
